src/function/scalar/list/list_zip.cpp

```diff
@@ -0,0 +1,157 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+
+namespace duckdb {
+
+static void ListZipFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	idx_t count = args.size();
+	idx_t args_size = args.ColumnCount();
+	auto *result_data = FlatVector::GetData<list_entry_t>(result);
+	auto &result_struct = ListVector::GetEntry(result);
+	auto &struct_entries = StructVector::GetEntries(result_struct);
+	bool flags_given = false;
+
+	// Check flag
```

Comment:
What is this flag doing? It is not completely clear to me.

Reply:
With the flag, you can decide if your result has the length of your longest given list or shortest. So you can determine if you want to cut off some values or extend with nulls. E.g.
`list_zip([1,2,3], [1], false)` to `[{v1: 1, v2: 1}]`
or
`list_zip([1,2,3], [1], true)` to `[{v1: 1, v2: 1}, {v1: 2, v2: NULL}, {v1: 3, v2: NULL}]`